crates/starknet-devnet-core/src/starknet/mod.rs

```diff
@@ -1164,7 +1102,16 @@ impl Starknet {
         let blockifier_transactions = {
             transactions
                 .iter()
-                .map(|txn| {
+                .enumerate()
+                .map(|(idx, txn)| {
+                    if txn.is_max_fee_zero_value() && !skip_fee_charge {
```

Comment:
Can you just add a comment here explaining that we must take care of this special case due to a shortcoming in blockifier (perhaps paste the slack link)?

Reply:
[2457229](https://github.com/0xSpaceShard/starknet-devnet-rs/pull/628/commits/2457229466dc7030b028922dc91057356fd1e2fe)